### PR TITLE
Regional languages support

### PR DESCRIPTION
**Pull request by sk-juroot**
## Description
Reworked `CustomActivity.setLocale()` to internally handle regional and non-regional language codes on various system versions.

## Related issues
- #329 

## Additional context
This code contains hardcoded check for Portuguese language (as the only regional-coded language in this project right now), since various Android devices may report this language as non-regional, causing falling back to English even when proper language asset is present.

### CODE REVIEW

**Comment by bekuno** [comment]
Ok to test

**Comment by Lineflyer** [comment]
@sk-juroot Your local branch also contains the commits already merged as part of #332. Can you rebase / recreate this PR as on top fixes based on current `master`?

**Comment by sk-juroot** [comment]
Sure, will open another PR from different branch to merge this change

File: src/main/java/menion/android/whereyougo/gui/extension/activity/CustomActivity.java [hunk 1]
```diff
@@ -253,14 +253,34 @@ private void unbindDrawables(View view) {
...
+        } else {
+            lang = new Locale(languageCode).toString();
+        }
+
+        // need to parse language codes with dash and hyphen
+        String[] loc = lang.split("[-_]");
+        if (loc.length == 1) {
+            if (loc[0].equals("pt")) {
+                // nasty hotfix for Portuguese language selected without regional code
+                locale = new Locale(loc[0], loc[0]);
```
Comment (bekuno) [inline src/main/java/menion/android/whereyougo/gui/extension/activity/CustomActivity.java:273]: Let's do it the way around - rename the language `pt_PT` to `pt`.
I created Issue #342 for it.
After that these lines are not more necessary.

**Pull request closed without merging**